### PR TITLE
Restore and use linesIterator to avoid conflict with JDK 11's String.lines

### PR DESCRIPTION
**Pull request by retronym**

### CODE REVIEW

**Comment by retronym** [comment]
Fixes scala/bug#11125

**Review by xuwei-k — APPROVED** [review]

File: src/library/scala/collection/StringOps.scala [hunk 1]
```diff
@@ -624,10 +624,20 @@ final class StringOps(private val s: String) extends AnyVal {
     }
   }
 
+  /** Return all lines in this string in an iterator, excluding trailing line
+    *  end characters; i.e., apply `.stripLineEnd` to all lines
+    *  returned by `linesWithSeparators`.
+    *
+    * This is an alias for `lines` with a name that won't conflict with `String.lines` added in JDK 11.
```
Comment (SethTisue) [inline src/library/scala/collection/StringOps.scala]: I don't think the wording here quite makes sense. `lines` is now deprecated, and it seems weird to me to describe the recommended method as an alias for a deprecated method.

I'd suggest simply omitting this sentence.

**Review by SethTisue — CHANGES_REQUESTED** [review]
one suggested change, otherwise LGTM for merge

File: src/library/scala/collection/StringOps.scala (hunk 1 above)
Comment (retronym) [reply src/library/scala/collection/StringOps.scala]: Good point, I've removed this.

**Review by SethTisue — APPROVED** [review]

**Pull request merged**